### PR TITLE
Create Travis integration for macOS and Linux

### PR DESCRIPTION
**Pull request by uudisaru**
TERA-54:
- Created build scripts for installing macOS dependencies
- Updated the macOS build instructions in README.md
- Created Travis configuration file, scripts for osx and Linux
- Deployment artefacts currently not archived

Signed-off-by: Toomas Uudisaru <toomas.uudisaru@aktors.ee>

### CODE REVIEW

File: macos-manual.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+### OSX
+
+#### 1. Install dependencies
+
+###### a) Build tools
+
+    install xcode
+    Qt5 https://www.qt.io/download-open-source/ TODO see back
+    TODO git
```
Comment (metsma) [inline macos-manual.md]: git is part of xcode

File: macos-manual.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+###### b) CMake
+
+Building from source https://cmake.org/files/v3.6/cmake-3.6.2.tar.gz
+
+    ./bootstrap
+    make
+    sudo make install
```
Comment (metsma) [inline macos-manual.md]: Why compile? there are official binaries available

File: macos-manual.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+Download and install the latest zlib from http://zlib.net/
+
+    mkdir -p ~/cmake_builds/ && cd ~/cmake_builds/
+    curl -O -L http://zlib.net/zlib-1.2.11.tar.gz
+    tar xf zlib-1.2.11.tar.gz
+    mkdir -p ~/cmake_builds/zlib && cd ~/cmake_builds/zlib
+    cmake -DCMAKE_INSTALL_PREFIX=~/cmake_builds/zlib_bin -G "Unix Makefiles" ../zlib-1.2.11
+    cmake --build .
+    make install
```
Comment (metsma) [inline macos-manual.md:23]: OSX has zlib libs and headers

File: macos-manual.md (hunk 1 above)
Comment (uudisaru) [reply macos-manual.md]: This file contains instructions which were in main README.md before, I have not changed instructions for manual install. I agree the instructions are not optimal, I shall remove unnecessary steps.

File: macos-manual.md (hunk 3 above)
Comment (uudisaru) [reply macos-manual.md:23]: Tera links zlib statically, that is why zlib is compiled: 
https://github.com/open-eid/TeRa/blob/master/CMakeLists.txt#L83

**Review by metsma — APPROVED** [review]
Question remains, why is static zlib required?

**Pull request merged**